sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5331,15 +5331,15 @@ pi_result piEventGetProfilingInfo(pi_event Event, pi_profiling_info ParamName,
   case PI_PROFILING_INFO_COMMAND_START: {
     ZE_CALL(zeEventQueryKernelTimestamp, (Event->ZeEvent, &tsResult));
     uint64_t ContextStartTime =
-        (tsResult.context.kernelStart & TimestampMaxValue) * ZeTimerResolution;
```

Comment:
any comments why using "context" is incorrect here? (the event being profiled belongs to a L0 context)

Reply:
Thanks. I don't see SYCL spec saying anything about what the timestamp values should include: https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_event_information_and_profiling_descriptors

So I guess it's OK to return either one. Just curious what was the problem while returning "context" timestamps?

Reply:
@smaslov-intel

Right, i didnt see anything in particular in SYCL specl.

Now, let's say we submit a kernel for execution, but because of scheduling and other hardware mechanisms, the kernel doesn't start execution right away. Now, it starts execution and gets preempted, and then resume again.

context timestamp would include only the time where the kernel was actively being executed, while the global timestamp would include even the times the kernel is waiting for active resources to execute. I would say that since SYCL is not clear which one is used, we could assume global is the one of importance here, as you said.